src/app/notifications/notification-action.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * An action for a button, dropdown, etc:
+ *
+ * id - Optional unique Id for the filter field, useful for comparisons
+ * isDisabled - set to true to disable the action
+ * isSeparator - set to true if this is a placehodler for a separator rather than an action
+ * name - The name of the action, displayed on the button
+ * title - Optional title, used for the tooltip
+ */
+export class NotificationAction {
+  id?: string;
+  isDisabled?: boolean = false;
+  isSeparator?: boolean = false;
+  name: string;
+  title?: string;
+  callback: NotificationActionCallback;
```

Comment:
If you want to do it with events, I would suggest that notificationService.message() should return an observable which is called when an action is invoked.

I can work what's needed in to widgets if you like, but we should move the APIs out of the library in that case.